pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -834,9 +835,11 @@ protected void internalSetAutoTopicCreation(AsyncResponse asyncResponse,
                         "Invalid configuration for autoTopicCreationOverride. the detail is "
                                 + validateResult.getErrorInfo());
             }
-            if (maxPartitions > 0 && autoTopicCreationOverride.getDefaultNumPartitions() > maxPartitions) {
-                throw new RestException(Status.NOT_ACCEPTABLE,
-                        "Number of partitions should be less than or equal to " + maxPartitions);
+            if (Objects.equals(autoTopicCreationOverride.getTopicType(), TopicType.PARTITIONED.toString())) {
+                if (maxPartitions > 0 && autoTopicCreationOverride.getDefaultNumPartitions() > maxPartitions) {
```

Comment:
if the problem is that autoTopicCreationOverride.getDefaultNumPartitions() can be null then we have to simply add a null check and fail the request if it is not passed while maxPartitions is > 0

Reply:
A simple null check cannot clearly express why it is null here, which will increase the burden on the maintainers later.
I think a better approach is to use logical judgments to avoid null checks everywhere.

Reply:
But you should add the null check anyway. Because here we can fail with NPE

Reply:
The null check already exists in bellow code:

https://github.com/mattisonchao/pulsar/blob/dec9c6bdc584bfbabe2120a681f090bfb622d030/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/AutoTopicCreationOverrideImpl.java#L39-L61